onc-healthit/lantern-back-end
# Parsing FHIR endpoint list

This ticket adds the ability to read in lists of endpoints formatted using FHIR, based on this: https://www.hl7.org/fhir/endpoint-examples-general-template.json.html
- I had to change how the list source was being referenced so that all FHIR endpoints would be processed the same way but would have their list URL as the list source value.

To test:
- The easiest way to test the FHIR endpoint list is to copy the example list of endpoints in `endpointmanager/resources/FHIREndpointSources.json` to the main `resources/dev_resources` directory and then either reference it directly in `populatedb.sh` like the CareEvolution list or add to the `resources/prod_resources/EndpointResourcesList.json`:
```
{
        "EndpointName":"FHIR",
        "FileName":"FHIREndpointSources.json",
        "URL": "www.testurl.com"
}
```
- The URL is optional, so make sure when checking the values in the `fhir_endpoints` database that the list source is either the URL you put in, or if you left it empty, `FHIRSource`
-----------------

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-287
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
  - Code accomplishes the tasks purpose.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.
